lib/DBIx/Class/ResultSource.pm

```diff
@@ -1531,7 +1531,7 @@ sub reverse_relationship_info {
 sub __strip_relcond {
   +{
     map
-      { map { /^ (?:foreign|self) \. (\w+) $/x } ($_, $_[1]{$_}) }
+      { map { /^ (?:foreign|self) \. (.+) $/x } ($_, $_[1]{$_}) }
```

Comment:
While this regex is not DBMS-specific all new allowed characters (all of `\W`) seem to be relevant for MS-SQL only. So I suggest to still limit the allowed characters to those allowed in MS-SQL.

I took a quick look at https://docs.microsoft.com/en-us/sql/odbc/microsoft/column-name-limitations?view=sql-server-2017

* `` ` `` is not allowed
* `|` is not allowed

So I suggest ``[^`|]+``.